compiler/src/dotty/tools/dotc/printing/RefinedPrinter.scala

```diff
@@ -126,8 +126,8 @@ class RefinedPrinter(_ctx: Context) extends PlainPrinter(_ctx) {
     def isInfixType(tp: Type): Boolean = tp match {
       case AppliedType(tycon, args) =>
         args.length == 2 &&
-          !Character.isUnicodeIdentifierStart(tycon.typeSymbol.name.toString.head)
-          // TODO: Once we use the 2.12 stdlib, also check the @showAsInfix annotation
+          !Character.isUnicodeIdentifierStart(tycon.typeSymbol.name.toString.head) &&
```

Comment:
This should be an `||` not an `&&` as we want infix types to be used either for type constructors which start with a symbol _or_ that have the `@showAsInfix` annotation.

Reply:
There's one more problem. I didnt notice that this annotation takes a boolean argument, which should be checked as well.